### PR TITLE
feat: Create mixpanel interface

### PR DESCRIPTION
**Pull request by camiloperezv**
This PR creates the interface to track events using mixpanel API

### CODE REVIEW

File: mixpanel.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+/******************************************************
+PRIVATE METHODS
+*******************************************************/
+// format all the arguments and token into the base64 argument for url data argument
+func formatArgumentToString(token string, event string, distinctID string, properties map[string]interface{}) (string, error) {
+	jsonArguments, err := json.Marshal(properties)
+	if err != nil || distinctID == "" || token == "" || event == "" {
+		return "", errors.New(errInvalidArguments)
```
Comment (hendrikvolschenk) [inline mixpanel.go]: `[q]` Is it not better to return `nil, errors.New(errInvalidArguments)` here?

File: mixpanel_test.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
+package mixpanel
+
+import (
+	"testing"
+)
+
+func TestTrackEvent(t *testing.T) {
+	testCases := []struct {
+		Name             string
+		HasError         bool
```
Comment (hendrikvolschenk) [inline mixpanel_test.go]: `[q]` I don't see this being used, or am I missing something?

File: mixpanel_test.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+				}
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			testCase.ValidateFunction(t)
+		})
+	}
+}
```
Comment (hendrikvolschenk) [inline mixpanel_test.go]: `[q]` I have a few questions about these tests, but maybe we could discuss them in person please?

File: mixpanel.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+	}
+	// Public interface where all the public methos exists
+	Public interface {
+		TrackEvent(event, string, distinctID string, properties map[string]interface{}) error
+	}
+)
+
+/******************************************************
+PRIVATE INTERFACE
+*******************************************************/
```
[the remark this answers is not in the document]
Comment (camiloperezv) [reply mixpanel.go]: no, and yes, is a private method but belongs to the interface, this method are private and can be accessed by both interfaces.

File: mixpanel.go (hunk 1 above)
Comment (camiloperezv) [reply mixpanel.go]: you can't return nil if the declared return value is a string, the empty (nil) value for string is ""

File: mixpanel_test.go (hunk 3 above)
Comment (camiloperezv) [reply mixpanel_test.go]: yes

File: mixpanel.go (hunk 1 above)
Comment (gillchristian) [reply mixpanel.go]: Go did this weird thing where they don't have `nil` for some types (yay) but have it for some others (wtf) :man_shrugging:

Comment (camiloperezv) [reply mixpanel.go]: I think just for errors and pointers (with pointers have sense)

Comment (hendrikvolschenk) [reply mixpanel.go]: https://www.reactiongifs.com/wp-content/uploads/2013/10/tom-delonge-wtf1.gif

Comment (gillchristian) [reply mixpanel.go]: Once you learn which one is which "make sense" but if you think about the reasons for not having `nil` on _some types_ you realize they apply for every type.

Even the author of `null` is not convinced about it (ref: https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/)

**Pull request merged**